pennylane/estimator/wires_manager.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""This module contains the base class for qubit management"""
+
+
+import pennylane as qml
+
+# pylint: disable= too-few-public-methods
+
+
+class _WireResourceManager:
+    r"""Manages and tracks the auxiliary and algorithmic qubits used in a quantum circuit.
+
+    Args:
+        work_wires (int or Dict[str, int]): Number of work wires or a dictionary containing
+            number of clean and dirty work wires. All ``work_wires`` are assumed to be clean when
+            `int` is provided.
+        algo_wires (int): Number of algorithmic wires, default value is ``0``.
+        tight_budget (bool): Determines whether extra clean qubits can be allocated when they
+            exceed the available amount. The default is ``False``.
+
+    **Example**
+
+    >>> q = _WireResourceManager(
+    ...             work_wires={"clean": 2, "dirty": 2},
+    ...             tight_budget=False,
+    ...     )
+    >>> print(q)
+    _WireResourceManager(clean=2, dirty=2, logic=0, tight_budget=False)
+
+    """
+
+    def __init__(self, work_wires: int | dict, algo_wires=0, tight_budget=False) -> None:
+
+        if isinstance(work_wires, dict):
+            clean_wires = work_wires["clean"]
+            dirty_wires = work_wires["dirty"]
+        else:
+            clean_wires = work_wires
+            dirty_wires = 0
+
+        self.tight_budget = tight_budget
+        self._logic_qubit_counts = algo_wires
+        self._clean_qubit_counts = clean_wires
+        self._dirty_qubit_counts = dirty_wires
```

Comment:
```suggestion
        self.algo_wires = algo_wires
        self.clean_wires = clean_wires
        self.dirty_wires = dirty_wires
```
and then you can get rid of the `@property` methods below. This is already an internal private class, if you're not doing anything special in the getters, you can safely make these attributes public.